tests/repo_finder/test_provenance_extractor.py

```diff
@@ -0,0 +1,457 @@
+                                    }
+                                },
+                                {
+                                    "uri": "git+https://github.com/oracle-samples/macaron@refs/heads/main"
+                                }
+                            ]
+                        }
+                    }
+                }
+            """
+
+
+@pytest.fixture(name="slsa_v02_provenance")
+def slsa_v02_provenance_() -> str:
+    """Return a valid SLSA v02 provenance."""
+    return """
+                {
+                    "_type": "https://in-toto.io/Statement/v0.1",
+                    "subject": [],
+                    "predicateType": "https://slsa.dev/provenance/v0.2",
+                    "predicate": {
+                        "invocation": {
+                            "configSource": {
+                                "uri": "git+https://github.com/oracle/macaron@refs/heads/staging",
+                                "digest": {
+                                    "sha1": "51aa22a42ec1bffa71518041a6a6d42d40bf50f0"
+                                }
+                            }
+                        }
+                    }
+                }
+            """
+
+
+@pytest.fixture(name="slsa_v01_provenance")
+def slsa_v01_provenance_() -> str:
+    """Return a valid SLSA v01 provenance."""
+    return """
+                {
+                    "_type": "https://in-toto.io/Statement/v0.1",
+                    "subject": [],
+                    "predicateType": "https://slsa.dev/provenance/v0.1",
+                    "predicate": {
+                        "recipe": {
+                            "definedInMaterial": 1
+                        },
+                        "materials": [
+                            {
+                                "uri": "git+https://github.com/oracle-samples/macaron@refs/heads/main"
+                            },
+                            {
+                                "uri": "git+https://github.com/oracle/macaron@refs/heads/main",
+                                "digest": {
+                                    "sha256": "51aa22a42ec1bffa71518041a6a6d42d40bf50f0"
+                                }
+                            }
+                        ]
+                    }
+                }
+            """
+
+
+@pytest.fixture(name="target_repository")
+def target_repository_() -> str:
+    """Return the target repository URL."""
+    return "https://github.com/oracle/macaron"
+
+
+@pytest.fixture(name="target_commit")
+def target_commit_() -> str:
+    """Return the target commit hash."""
+    return "51aa22a42ec1bffa71518041a6a6d42d40bf50f0"
+
+
+def test_slsa_v1_gcb_1(slsa_v1_gcb_1_provenance: str, target_repository: str, target_commit: str) -> None:
+    """Test SLSA v1 provenance with build type gcb and sourceToBuild."""
+    payload = json.loads(slsa_v1_gcb_1_provenance)
+    assert isinstance(payload, dict)
```

Comment:
I think this deserialization should be done within the fixture function, not within the test function.